onnxruntime/python/tools/transformers/optimizer.py

```diff
@@ -252,6 +253,9 @@ def optimize_model(
     # stable.
     disabled_optimizers = ["ConstantSharing"]
     temp_model_path = None
+    temp_dir = tempfile.TemporaryDirectory()
+    optimized_model_name = "{}_o{}_{}.onnx".format(input[:-5], opt_level, "gpu" if use_gpu else "cpu")
```

Comment:
input is a path so input[:-5] might not be a proper filename. Since it is in a random temp directory, we can use a simple name like "model_o{}_{}.onnx".

Reply:
fixed. thanks